### PR TITLE
Fix issue #24 to return BadRequest for empty body

### PR DESCRIPTION
**Pull request by zhang-hua**
# Type of change? 

- [x] Bug fix
- [ ] New feature
- [ ] Enhancement
- [ ] Breaking change (breaks backward compatibility)

### CODE REVIEW

File: app/com/microsoft/azure/iotsolutions/uiconfig/webservice/v1/controllers/DeviceGroupController.java [hunk 1]
```diff
@@ -34,13 +36,21 @@ public DeviceGroupController(IStorage storage) {
         return storage.getDeviceGroupAsync(id).thenApplyAsync(m -> ok(toJson(new DeviceGroupApiModel(m))));
     }
 
-    public CompletionStage<Result> createAsync() throws BaseException {
-        DeviceGroupApiModel input = Json.fromJson(request().body().asJson(), DeviceGroupApiModel.class);
+    public CompletionStage<Result> createAsync() throws BaseException, BadRequestException {
+        JsonNode json = request().body().asJson();
+        if(json == null || json.size() == 0) {
```
Comment (dluc) [inline app/com/microsoft/azure/iotsolutions/uiconfig/webservice/v1/controllers/DeviceGroupController.java]: code style: add a " " after "if"

File: app/com/microsoft/azure/iotsolutions/uiconfig/webservice/v1/controllers/DeviceGroupController.java [hunk 2]
```diff
@@ -34,13 +36,21 @@ public DeviceGroupController(IStorage storage) {
...
+        }
+        DeviceGroupApiModel input = Json.fromJson(json, DeviceGroupApiModel.class);
         return storage.createDeviceGroupAsync(input.ToServiceModel()).thenApplyAsync(m -> ok(toJson(new DeviceGroupApiModel(m))));
     }
 
-    public CompletionStage<Result> updateAsync(String id) throws BaseException {
-        DeviceGroupApiModel input = Json.fromJson(request().body().asJson(), DeviceGroupApiModel.class);
+    public CompletionStage<Result> updateAsync(String id) throws BaseException, BadRequestException {
+        JsonNode json = request().body().asJson();
+        if(json == null || json.size() == 0) {
```
Comment (dluc) [inline app/com/microsoft/azure/iotsolutions/uiconfig/webservice/v1/controllers/DeviceGroupController.java]: code style

File: app/com/microsoft/azure/iotsolutions/uiconfig/webservice/v1/controllers/DeviceGroupController.java (hunk 1 above)
Comment (zhang-hua) [reply app/com/microsoft/azure/iotsolutions/uiconfig/webservice/v1/controllers/DeviceGroupController.java]: fixed.

File: app/com/microsoft/azure/iotsolutions/uiconfig/webservice/v1/controllers/DeviceGroupController.java (hunk 2 above)
Comment (zhang-hua) [reply app/com/microsoft/azure/iotsolutions/uiconfig/webservice/v1/controllers/DeviceGroupController.java]: fixed.

**Pull request merged**